guides/documentations/essentials.md

```diff
@@ -0,0 +1,209 @@
+- `isParsable`: Indicate if this parser can parse this file. Return `this.__parser.isParsable()`.
+- `getModels`: Get models from all project files. Return `this.__parser.getModels(files.filter((file) => this.isParsable(file))`.
+- `render`: Return all generated files from plugin data. Configuration file is used for saving the components' configuration. Call `this.__renderer.renderConfiguration()` and return `this.__renderer.render().concat(configurationFile)`.
+- `arrangeComponentsPosition`: Reorganize nodes layout algorithmically.
+- `repositionComponent`: Reposition a component where there is room for it.
+- `getComponentPosition`: Get the coordinates to use for component draw options.
+- `addComponent`: Create new component with draw option if passed as param. Return `this.data.addComponent()`.
+
+## Metadata management
+
+A component represents a piece of code, related to a specific service, that is defined via the Metadata. It is the building block for creating a diagram. Some components are connected to each other either by links, or because one component may contain another. These actions can be performed through the 'draw' interface or by modifying component attributes in the component panel (in Leto-modelizer).
+
+Metadata is used to list all components of the application, define their content, how they are displayed and specify the rules for their various properties.
+Here is how you can initialize a component:
+
+```js
+new Component({
+  id: 'id',
+  name: 'name',
+  definition: new ComponentDefinition(),
+  drawOption: new ComponentDrawOption(),
+  attributes: [new ComponentAttribute()],
+})
+```
+
+To make your plugin work, we need to have an array of [ComponentDefinitions](https://github.com/ditrit/leto-modelizer-plugin-core/blob/main/src/models/ComponentDefinition.js).
+Component definitions are loaded from a metadata file. Each component definition has:
+- A name
+- An icon
+- Attributes
+- ...
+
+| Left: Components list in Leto-modelizer - Middle: Related Metadata - Left: Component Panel |
+| :--: |
+| <img height="400" src="../../documentations/metadata.png"/> |
+
+Attributes of a component represent all the related properties that can/must be set inside the Component Panel (Right side of the previous image).
+Link Definitions can also be loaded from the metadata file (for colored links, dashed links...).
+
+Once the component definitions are loaded, components can be instantiated from the components list (Left side of the previous image) in Leto-modelizer.
+The DefaultMetadata class provides two methods:
+
+- `validate`: Validate the metadata.
+- `parse`: Set all component/link definitions in PluginData from your provided metadata.
+
+## Renderer
+
+The Renderer is used to serialize all the instantiated components into source files (`.tf` files for Terraform, `.yaml` files for Kubernetes, ...)
+It uses Jinja/Nunjucks templates.
```

Comment:
depend of plugin do same think like Parser.

Reply:
done